llvm/lib/Target/AArch64/SMEInstrFormats.td

```diff
@@ -1965,21 +1987,91 @@ class sme2_mla_long_array_index_base<bits<2> op0, bits<2> op, Operand index_ty,
   let Inst{20}    = !if(!eq(vg_acronym, ""), 0, 1);
   let Inst{19-16} = Zm;
   let Inst{14-13} = Rv;
-  let Inst{12}    = 0b1;
-  let Inst{4-3}   = op;
 
   let Constraints = "$ZAda = $_ZAda";
 }
 
-multiclass sme2_mla_long_array_index<string mnemonic, bits<2> op0, bits<2> op, ValueType zpr_ty, SDPatternOperator intrinsic> {
-  def _HtoS : sme2_mla_long_array_index_base<op0, op, uimm3s2range, ZPR16,
-                                          mnemonic>, SMEPseudo2Instr<NAME # _HtoS, 1> {
-    bits<3> i3;
+multiclass sme2_mla_long_array_index_za16<string mnemonic, SDPatternOperator intrinsic> {
```

Comment:
I am not sure why you did the refactoring this way.
It is possible to review, but it is quite confusing to check. I would not like to trust only in the tests.
Is it possible to no mix the refactoring with the intrinsic implementation?

Reply:
That's fine, I'll save it for a another time :+1: .